### PR TITLE
Add persistent queue to usage collector

### PR DESCRIPTION
**Pull request by grzegorz-roboflow**
# Description

Add persistent queue to usage collector, to keep payloads when device is offline

## Type of change

Please delete options that are not relevant.

-   [x] New feature (non-breaking change which adds functionality)

## How has this change been tested, please provide a testcase or example of how you tested the change?

Unit tests added

## Any specific deployment considerations

N/A

## Docs

N/A

### CODE REVIEW

File: inference/usage_tracking/persistent_queue.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import time
+
+from typing_extensions import Any, Dict, List
+
+from inference.core.env import MODEL_CACHE_DIR
+from inference.core.logger import logger
+
+
+class PersistentQueue:
+    def __init__(self, db_file_path: str = os.path.join(MODEL_CACHE_DIR, "usage.db")):
```
Comment (PawelPeczek-Roboflow) [inline inference/usage_tracking/persistent_queue.py:13]: maybe if we have cls method init that would initialise everything and pass connection to ready DB we could:
* avoid potential exceptions from constructor - which is just my personal preference
* have an easy way to mock connections in tests

File: inference/usage_tracking/persistent_queue.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import json
```
Comment (PawelPeczek-Roboflow) [inline inference/usage_tracking/persistent_queue.py:1]: general question: could we estimate performance implications - from what I remember we will be using that on bg thread only once for a while, compressing messages avoiding excessive growth, is that right?

**Review by PawelPeczek-Roboflow — DISMISSED** [review]

File: inference/usage_tracking/persistent_queue.py (hunk 1 above)
Comment (grzegorz-roboflow) [reply inference/usage_tracking/persistent_queue.py:12]: to mock `db_file_path` is set to ":memory:", I added some tests covering this class, please have a look.

Also, instantiate this class in UsageCollector with catch-them-all exception handling so UsageCollector falls back to regular Queue - if `os.path.join` fails it's caught in upper layer and graceful fall-back is applied

File: inference/usage_tracking/persistent_queue.py (hunk 2 above)
Comment (grzegorz-roboflow) [reply inference/usage_tracking/persistent_queue.py:1]: yes, we are dumping usage payload to the queue every configurable timeout (currently 10 seconds), and it's done outside of the main thread

**Review by PawelPeczek-Roboflow — APPROVED** [review]

**Review by PawelPeczek-Roboflow — APPROVED** [review]

**Review by PawelPeczek-Roboflow — APPROVED** [review]

**Review by PawelPeczek-Roboflow — APPROVED** [review]

**Pull request merged**